modules/affile/affile-source.c

```diff
@@ -272,7 +309,8 @@ affile_sd_notify(LogPipe *s, gint notify_code, gpointer user_data)
         log_pipe_deinit((LogPipe *) self->reader);
         log_pipe_unref((LogPipe *) self->reader);
         self->reader = NULL;
-        
+        affile_sd_reset_file_state(cfg->state, affile_sd_format_persist_name(self));
+
         if (affile_sd_open_file(self, self->filename->str, &fd))
```

Comment:
I don't get why we need to drop the file state in case it was moved. This causes entries in the persist-state file to profilerate, leaking persist state storage.